src/abstraction/ak.rs

```diff
@@ -0,0 +1,197 @@
+            | SignatureScheme::Sm2 => Ok(()),
+            _ => Err(Error::local_error(WrapperErrorKind::InconsistentParams)),
+        },
+    }
+}
+
+fn create_ak_public(
+    key_alg: AsymmetricAlgorithm,
+    hash_alg: HashingAlgorithm,
+    sign_alg: SignatureScheme,
+) -> Result<TPM2B_PUBLIC> {
+    let mut obj_attrs = ObjectAttributes(0);
+    obj_attrs.set_restricted(true);
+    obj_attrs.set_user_with_auth(true);
+    obj_attrs.set_sign_encrypt(true);
+    obj_attrs.set_decrypt(false);
+    obj_attrs.set_fixed_tpm(true);
+    obj_attrs.set_fixed_parent(true);
+    obj_attrs.set_sensitive_data_origin(true);
+
+    match key_alg {
+        AsymmetricAlgorithm::Rsa => Tpm2BPublicBuilder::new()
+            .with_type(TPM2_ALG_RSA)
+            .with_name_alg(hash_alg.into())
+            .with_object_attributes(obj_attrs)
+            .with_parms(PublicParmsUnion::RsaDetail(TPMS_RSA_PARMS {
+                symmetric: TPMT_SYM_DEF_OBJECT {
+                    algorithm: TPM2_ALG_NULL,
+                    keyBits: TPMU_SYM_KEY_BITS { aes: 0 },
+                    mode: TPMU_SYM_MODE { aes: TPM2_ALG_NULL },
+                },
+                scheme: TPMT_RSA_SCHEME {
+                    scheme: sign_alg.into(),
+                    details: TPMU_ASYM_SCHEME {
+                        anySig: TPMS_SCHEME_HASH {
+                            hashAlg: hash_alg.into(),
+                        },
+                    },
+                },
+                keyBits: 2048,
+                exponent: 0,
+            }))
+            .with_unique(PublicIdUnion::Rsa(Box::new(Default::default()))),
+        AsymmetricAlgorithm::Ecc => Tpm2BPublicBuilder::new()
+            .with_type(TPM2_ALG_ECC)
+            .with_name_alg(hash_alg.into())
+            .with_object_attributes(obj_attrs)
+            .with_parms(PublicParmsUnion::EccDetail(TPMS_ECC_PARMS {
+                symmetric: TPMT_SYM_DEF_OBJECT {
+                    algorithm: TPM2_ALG_NULL,
+                    keyBits: TPMU_SYM_KEY_BITS { sym: 0 },
+                    mode: TPMU_SYM_MODE { sym: TPM2_ALG_NULL },
+                },
+                scheme: TPMT_ECC_SCHEME {
+                    scheme: TPM2_ALG_NULL,
+                    details: TPMU_ASYM_SCHEME {
+                        anySig: TPMS_SCHEME_HASH {
+                            hashAlg: hash_alg.into(),
+                        },
+                    },
+                },
+                curveID: TPM2_ECC_NIST_P256,
+                kdf: TPMT_KDF_SCHEME {
+                    scheme: TPM2_ALG_NULL,
+                    details: Default::default(),
+                },
+            }))
+            .with_unique(PublicIdUnion::Ecc(Box::new(Default::default()))),
+    }
+    .build()
+}
+
+pub fn load_ak(
+    context: &mut Context,
+    parent: KeyHandle,
+    ak_auth_value: Option<&Auth>,
+    private: Private,
+    public: TPM2B_PUBLIC,
+) -> Result<KeyHandle> {
+    let session = match context.start_auth_session(
```

Comment:
Do you need to flush this session before returning? It would be flushed when the `Context` gets dropped, but still